nextcord/automod.py

```diff
@@ -0,0 +1,351 @@
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+"""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, overload
+
+from .enums import ActionType, EventType, KeywordPresetType, TriggerType, try_enum
+from .mixins import Hashable
+from .utils import MISSING, snowflake_time
+
+if TYPE_CHECKING:
+    from datetime import datetime
+    from typing import List, Optional
+
+    from .abc import GuildChannel
+    from .errors import InvalidArgument
+    from .guild import Guild
+    from .member import Member
+    from .role import Role
+    from .state import ConnectionState
+    from .types.automod import (
+        ActionMetadata as ActionMetadataPayload,
+        AutoModerationAction as AutoModerationActionPayload,
+        AutoModerationRule as AutoModerationRulePayload,
+        TriggerMetadata as TriggerMetadataPayload,
+    )
+
+
+__all__ = ("AutoModerationRule", "AutoModerationAction")
+
+
+class AutoModerationAction:
+    """Represent an auto moderation action.
+
+    Attributes
+    -----------
+    type: :class:`ActionType`
+        The action type of this action.
+    notify_channel_id: Optional[:class:`int`]
+        The ID of the channel that this action will notify in.
+    timeout_seconds: Optional[:class:`int`]
+        The number of seconds this rule should timeout when someone breaks.
+    """
+
+    def __init__(self, data: AutoModerationActionPayload):
+        self.type: ActionType = try_enum(ActionType, data["type"])
+        self.notify_channel_id: Optional[int] = None
+        self.timeout_seconds: Optional[int] = None
+        if data.get("metadata"):
+            self._unpack_metadata(data["metadata"])  # type: ignore
+
+    def __repr__(self):
+        attrs = (
+            ("type", self.type),
+            ("notify_channel_id", self.notify_channel_id),
+            ("timeout_seconds", self.timeout_seconds),
+        )
+        inner = " ".join("%s=%r" % t for t in attrs)
+        return f"<AutoModerationAction {inner}>"
+
+    def _unpack_metadata(self, action_metadata: ActionMetadataPayload):
+
+        if action_metadata.get("channel_id") is not None:
+            self.notify_channel_id: Optional[int] = int(action_metadata.get("channel_id"))
+        if action_metadata.get("duration_seconds") is not None:
+            self.timeout_seconds: Optional[int] = action_metadata.get("duration_seconds")
+
+
+class AutoModerationRule(Hashable):
+    """Represents an auto moderation rule.
+
+    .. container:: operations
+        .. describe:: x == y
+            Check if two auto moderation rule are equal.
+        .. describe:: x != y
+            Check if two auto moderation rule are not equal.
+
+    Attributes
+    -----------
+    id: :class:`int`
+        The auto moderation rule's ID.
+    guild: :class:`Guild`
+        The guild that this auto moderation rule active in.
```

Comment:
```
The guild that this auto moderation rule is active in.
```

Reply:
This comment has clearly not been fixed, why is it marked as resolved?

Reply:
```suggestion
        The guild that this auto moderation rule is active in.
```